app/controllers/admin/site_settings_controller.rb

```diff
@@ -37,127 +40,124 @@ def update
 
     id = new_setting_name if new_setting_name
 
-    raise_access_hidden_setting(id)
-
-    case SiteSetting.type_supervisor.get_type(id)
-    when :integer
-      value = value.tr("^-0-9", "")
-    when :file_size_restriction
-      value = value.tr("^0-9", "").to_i
-    when :uploaded_image_list
-      value = value.blank? ? "" : Upload.get_from_urls(value.split("|")).to_a
-    end
-
-    value = Upload.get_from_url(value) || "" if SiteSetting.type_supervisor.get_type(id) == :upload
-
-    update_existing_users = params[:update_existing_user].present?
     previous_value = value_or_default(SiteSetting.get(id)) if update_existing_users
 
-    SiteSetting.set_and_log(id, value, current_user)
-
-    if update_existing_users
-      new_value = value_or_default(value)
-
-      if (user_option = user_options[id.to_sym]).present?
-        if user_option == "text_size_key"
-          previous_value = UserOption.text_sizes[previous_value.to_sym]
-          new_value = UserOption.text_sizes[new_value.to_sym]
-        elsif user_option == "title_count_mode_key"
-          previous_value = UserOption.title_count_modes[previous_value.to_sym]
-          new_value = UserOption.title_count_modes[new_value.to_sym]
-        end
+    with_service(UpdateSiteSetting, setting_name: id, new_value: value, current_user:) do
```

Comment:
You don't need to pass in `current_user`. The `with_service` helper injects the controller context into the service when it runs, so you will have `current_user` and `guardian` automatically available. If your params were the same ones as `@params[:blah]` they would automatically get passed in too and you could do `with_service(UpdateSiteSetting) do`